### PR TITLE
Discard session API and bindings for `Room` 

### PR DESCRIPTION
**Pull request by BillCarsonFr**
- [x] Public API changes documented in changelogs (optional)

Signed-off-by:

### CODE REVIEW

File: crates/matrix-sdk/CHANGELOG.md [hunk 1]
```diff
@@ -54,6 +54,8 @@ Additions:
 - Add `Client::subscribe_to_room_updates` and `room::Common::subscribe_to_updates`
 - Add `Client::rooms_filtered`
 - Add methods on `Client` that can handle several authentication APIs.
+- Add new method `force_discard_session` on `Room` that allows to discard the current
```
Comment (Hywan) [inline crates/matrix-sdk/CHANGELOG.md:57]:
```suggestion
- Add new method `discard_room_keys` on `Room` that allows to discard the current
```

**Review by Hywan — CHANGES_REQUESTED** [review]
Thanks for the PR. Why is the name different in `matrix-sdk-ffi`?
Also, is it possible to get tests or do you feel that's not mandatory?

**Comment by Hywan** [comment]
I prefer to keep the same name amongst the crates. We don't really want to mess with emote names 🙂.

**Review by bnjbvr — CHANGES_REQUESTED** [review]
Can you add a test in the SDK please?

**Comment by poljar** [comment]
What's the state of this? Is this still needed?

**Comment by poljar** [comment]
> I prefer to keep the same name amongst the crates. We don't really want to mess with emote names 🙂.

I agree, the names in the public API of the SDK should use the `room` terminology, the bindings should just re-expose what the SDK does and call it the same.

**Review by poljar — APPROVED** [review]
There were still some small issues, but I dragged this over the finish line.

**Pull request merged**